File: apps/api/internal/models/vaccine_application.go
package models

import "time"

// VaccineApplication representa la aplicación de una dosis de vacuna o desparasitación.
// Está vinculado a un health_record que contiene la información general del tratamiento.
// El campo application_date indica cuándo se aplicó la dosis.
// Notes permite almacenar observaciones sobre la aplicación (lote, veterinario, etc.).
type VaccineApplication struct {
	ID             string     `json:"id"             gorm:"primaryKey;type:uuid;default:gen_random_uuid()"`
	HealthRecordID string     `json:"health_record_id" gorm:"type:uuid;not null;index"`
	ApplicationDate time.Time  `json:"application_date" gorm:"type:date;not null"`
	Notes          *string    `json:"notes"          gorm:"type:text;default:null"`
	CreatedAt      time.Time  `json:"created_at"     gorm:"type:timestamptz;not null;default:now()"`

	// Relación con HealthRecord — CASCADE: si se elimina el registro, se eliminan las aplicaciones.
	HealthRecord HealthRecord `json:"health_record" gorm:"foreignKey:HealthRecordID;constraint:OnDelete:CASCADE"`
}

File: apps/web/src/types/vaccineApplication.ts
import type { ApiResponse } from '@/types/shared'

/**
 * Aplicación de vacuna/desparasitación.
 * Representa una dosis aplicada de un tratamiento.
 */
export interface VaccineApplication {
  id: string
  health_record_id: string
  application_date: string
  notes: string | null
  created_at: string
}

/**
 * Payload para crear una aplicación de vacuna.
 */
export interface CreateVaccineApplicationPayload {
  health_record_id: string
  application_date: string
  notes?: string
  next_dose_date?: string | null
}

/**
 * Payload para actualizar una aplicación de vacuna.
 */
export interface UpdateVaccineApplicationPayload {
  application_date?: string
  notes?: string
}

// Re-exportar ApiResponse para compatibilidad
export type { ApiResponse }
